### ISSUE TITLE
User defined literals are crashing GCC IntelliSense

### ISSUE THREAD

**Issue by Novakov** [microsoft/vscode-cpptools#3481]
**Type: LanguageService**

**Describe the bug**
- OS and Version: Windows 10 x64 1809
- VS Code Version: 1.33.1
- C/C++ Extension Version: 0.22.1 (0.22.2-insiders also affected)
- Other extensions you installed (and if the issue persists after disabling them): None
- Using user defined integer literal used in function call when using GCC mode causes crash of `Microsoft.VSCode.CPP.IntelliSense.Msvc.exe`.

**To Reproduce**
1. Clone repository https://github.com/Novakov/vscode-cpp-literals-crash
2. Open cloned repository in VS Code
3. Adjust `compilerPath` in `.vscode/c_cpp_properties.json` to point to GCC compiler (issue confirmed on arm-none-eabi-gcc and MinGW)
4. Open C++ log panel
5. Open test.cpp
6. Move cursor over `Do` function call in line 5
7. IntelliSense crashes with error `Failed to read response from server: 4`

**Expected behavior**
Intellisense should not crash.

**Screenshots**
![2019-04-17_19-13-01](https://user-images.githubusercontent.com/420393/56307855-2356b200-6146-11e9-85f8-135fcdbdf356.gif)

**Additional context**
C++ log output (limited to the moment of crash)
```
textDocument/hover: 43
Failed to read response from server: 4
IntelliSense engine is not responding. Using the Tag Parser instead.
using Tag Parser for quick info
IntelliSense engine is not responding. Using the Tag Parser instead.
IntelliSense engine is not responding. Using the Tag Parser instead.
Checking for syntax errors: file:///d%3A/tmp/vscode/literals-crash/test.cpp
IntelliSense client is currently disconnected
```

`Microsoft.VSCode.CPP.IntelliSense.Msvc.exe` memory dump: [Microsoft.VSCode.CPP.IntelliSense.Msvc.exe.zip](https://github.com/Microsoft/vscode-cpptools/files/3090786/Microsoft.VSCode.CPP.IntelliSense.Msvc.exe.zip)

Call stack:
![image](https://user-images.githubusercontent.com/420393/56308323-55b4df00-6147-11e9-90ff-17d506b55c9f.png)

* User defined literals using parameter instead of template are working fine (`constexpr int operator"" _my( unsigned long long n) { return 0; }`)
* MSVC IntelliSense mode is not crashing

**Comment by sean-mcmanus** [comment]
Thanks for reporting this. The bug is from VS 2019 -- I've filed a bug at https://developercommunity.visualstudio.com/content/problem/537614/user-defined-literals-crash-c-intellisense-with-in.html . You can login with your GitHub account to upvote that issue if you want.

**Comment by sean-mcmanus** [comment]
Fixed with 0.24.0-insiders (available by setting "C_Cpp.updateChannel" to "Insiders").

**Comment by Novakov** [comment]
Seems to be working fine :)
